### PR TITLE
Overhaul of plotting, support for SOS and more

### PR DESCRIPTION
**Pull request by Dvermetten**
This pull request contains many small updates, with the following major changes:
- Changed plotting functionality to use a common function for most plots, based on data.tables. This significantly increases performance, since the data generation and plotting can now be done separately, allowing for re-using relevant data where possible.
- Support for the SOS-based data (https://zenodo.org/record/3237024), including a parallel coordinate plot for final positions.
- Better costumization options for the plots.
- Improved handling of custom targets (ECDF and comparison across functions)
- Other minor updates (See commits included in this pull request)

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -42,7 +42,8 @@ Imports:
   igraph,
   scales,
   kableExtra,
-  PlayerRatings
+  PlayerRatings,
+  stringi
```
[the remark this answers is not in the document]
Comment (Dvermetten) [reply DESCRIPTION]: Yes, that package seemed like it was the easiest to use for things like substring detection

File: NAMESPACE [hunk 2]
```diff
@@ -190,9 +203,13 @@ importFrom(stats,p.adjust)
 importFrom(stats,quantile)
 importFrom(stats,rgeom)
 importFrom(stats,sd)
+importFrom(stringi,stri_detect_fixed)
```
[the remark this answers is not in the document]
Comment (Dvermetten) [reply NAMESPACE]: This file is auto-generated, so we shouldn't change this at all

File: R/DataSet.R [hunk 3]
```diff
@@ -171,18 +171,20 @@ DataSet <- function(info, verbose = F, maximization = NULL, format = IOHprofiler
...
+    if (!any((class(ds)) == 'DataSet'))
       stop("Operation only possible when all arguments are DataSets")
   }
   
+  fixed_attrs <- c('suite', 'maximization', 'DIM', 'funcId', 'algId', 'format')
+  
   info <- list()
-  for (attr_str in c('suite', 'maximization', 'DIM', 'funcId', 'algId')) {
+  for (attr_str in fixed_attrs) {
     temp  <-
```
[the remark this answers is not in the document]
Comment (Dvermetten) [reply R/DataSet.R]: Sure, I will change this

File: R/readFiles.R [hunk 4]
```diff
@@ -120,6 +120,80 @@ read_index_file__IOH <- function(fname) {
   data
 }
 
+#' Read single-objective COCO-based .info files and extract information
+#'
+#' @param fname The path to the .info file
+#' @return The data contained in the .info file
+#' @noRd
+read_index_file__COCO <- function(fname) {
```
[the remark this answers is not in the document]
Comment (Dvermetten) [reply R/readFiles.R]: It seems like the original version of the reading of coco's index files isn't there anymore, only the biobjective one. I have used that one as the basis for this function.

File: R/DataSet.R [hunk 5]
```diff
@@ -193,38 +195,50 @@ c.DataSet <- function(...) {
...
+  #TODO: Deal with cases where aligned parameters are present in original DataSets
   PAR <- list(
     'by_FV' = RT[names(RT) != 'RT'],
     'by_RT' = FV[names(FV) != 'FV']
   )
   
+  #Unaligned parameters
+  for (par_name in names(dsl[[1]]$PAR)) {
+    if (!par_name %in% c('by_FV', 'by_RT'))
+      PAR[[par_name]] <- unlist(lapply(dsl, function(x) {x$PAR[[par_name]]}), recursive = F)
```
[the remark this answers is not in the document]
Comment (Dvermetten) [reply R/DataSet.R]: These are currently specific to the files from Anna, so they include things like the final position and boundary corrections, of which we have just one value for each run.

File: R/stats.R [hunk 6]
```diff
@@ -474,16 +443,16 @@ get_default_ECDF_targets <- function(data, format_func = as.integer) {
 #' @export
 #' @examples 
 #' get_target_dt(dsl)
-get_target_dt <- function(dsList, which = 'by_FV') {
+get_target_dt <- function(dsList, which = 'by_RT') {
```
[the remark this answers is not in the document]
Comment (Dvermetten) [reply R/stats.R]: This was indeed confusing, and should probably be renamed for clarity. Maybe something like fixed-target and fixed-budget would be clearer?

File: R/stats.R [hunk 7]
```diff
@@ -102,10 +101,25 @@ pairwise.test.list <- function(x, max_eval, bootstrap.size = 30, ...) {
         x1 <- bootstrap_RT(x[[i]], max_eval[[i]], bootstrap.size)
         x2 <- bootstrap_RT(x[[j]], max_eval[[j]], bootstrap.size)
       }
-      options(warn = -1)
-      p.value[i, j] <- ks.test(x1, x2, alternative = 'greater', exact = F)$p.value
-      p.value[j, i] <- ks.test(x1, x2, alternative = 'less', exact = F)$p.value
-      options(warn = 0)
+      if (all(is.na(x1))) {
```
[the remark this answers is not in the document]
Comment (Dvermetten) [reply R/stats.R]: If one of them is all NA's it means, that no runs hit the target, so it has infinite ERT. Isn't it then the case that the other is better by definition?

File: NAMESPACE (hunk 2 above)
[the remark this answers is not in the document]
Comment (wangronin) [reply NAMESPACE]: Ok.. it seems no spacing after commas is their default coding style

**Pull request merged**